busypie/awaiter.py

```diff
@@ -39,11 +47,19 @@ def _raise_exception_if_not_ignored(self, e: Exception):
                 (ignored_exceptions and e.__class__ not in ignored_exceptions):
             raise e
 
-    def _validate_wait_constraint(self, condition_evaluator: ConditionEvaluator, start_time: float):
-        if (time.time() - start_time) > self._condition.wait_time_in_secs:
+    def _validate_condition_based_constraint(self, condition_evaluator: ConditionEvaluator, condition: bool):
+        if condition:
             raise busypie.ConditionTimeoutError(
                 self._describe(condition_evaluator), self._condition.wait_time_in_secs) from self._last_error
 
+    def _validate_lower_bound_wait_constraint(self, condition_evaluator: ConditionEvaluator, start_time: float):
```

Comment:
To get rid of this duplication you can create a partial and just pass it the appropriate condition. So you'll have something like this: ```validate_wait_bound(max_wait_excceeded())```

Reply:
no need using partial, pls see my solution